odata-csdl/14 Vocabulary and Annotation.md

```diff
@@ -1204,26 +1204,25 @@ Example ##ex:
 ## ##subsec Dynamic Expression
 
 Dynamic expressions allow assigning a calculated value to an applied
-term.
+term or term property.
 
 ### ##subsubsec Path Expressions
 
 Path expressions allow assigning a value to an applied term or term
 component. There are two kinds of path expressions:
-- A *model path* is used within
+- A *model path* is evaluated on the entity model of a service and resolves to a model element.
+Model paths are used within
 [Annotation Path](#AnnotationPath), [Model Element Path](#ModelElementPath),
-[Navigation Property Path](#NavigationPropertyPath), and [Property
-Path](#PropertyPath) expressions to traverse the model of a service and
-resolves to the model element identified by the path. It allows
-assigning values to terms or term properties of the [built-in
-types](#BuiltInTypesfordefiningVocabularyTerms) `Edm.AnnotationPath`,
-`Edm.NavigationPropertyPath`, `Edm.PropertyPath`, and their base types
-`Edm.AnyPropertyPath` and `Edm.ModelElementPath`.
-- An *instance path* is used within a
-[Value Path](#ValuePath) expression to traverse a graph of type
-instances and resolves to the value identified by the path. It allows
-assigning values to terms or term properties of built-in types other
-than the `Edm.*Path` types, or of any model-defined type.
+[Navigation Property Path](#NavigationPropertyPath), and [Property Path](#PropertyPath) expressions.
+They allow assigning values to terms or term properties of the 
+[built-in types](#BuiltInTypesfordefiningVocabularyTerms) `Edm.AnnotationPath`,
+`Edm.ModelElementPath`, `Edm.NavigationPropertyPath`, `Edm.PropertyPath`, and
+`Edm.AnyPropertyPath`.
+- An *instance path* is evaluated on a type instance and its nested or related type instances
```

Comment:
It can also mean primitive type instance, including instance of a type definition.

Reply:
~~~mermaid
classDiagram
  type instance <|-- structured type instance
  structured  type instance <|-- entity type instance
  structured type instance <|-- complex type instance
  type instance <|-- primitive type instance
~~~

Reply:
Can you give an example of an "instance of a type definition"? The `TypeDefinition` model element cannot be meant, because an instance path must evaluate to a value, not a model element.

Reply:
An instance of complex type `Size` from example 29 in [11 Type Definition](https://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/odata-csdl-xml-v4.01.html#sec_TypeDefinition):

~~~json
{ "Height": 10, "Weight": 3 }
~~~

The instance path `Height` resolves to the value `10`, which is an instance of type definition `Length` and ultimately an instance of `Edm.Int32`.

Reply:
I have edited your diagram. Is this correct?

Reply:
To be complete we could add enum type instance as another specialization of primitive type instance.